extensions/wikia/UserRenameTool/process/ProcessBase.class.php

```diff
@@ -0,0 +1,556 @@
+<?php
+
+namespace UserRenameTool\Process;
+
+use Wikia\Logger\Loggable;
+use Wikia\Logger\WikiaLogger;
+
+class ProcessBase {
+	use Loggable;
+
+	const FLAG_RENAME_DATA = 'renameData';
+	const RENAME_TAG = 'renamed_to';
+	const PROCESS_TAG = 'rename_in_progress';
+	const PHALANX_BLOCK_TAG = 'phalanx_block_id';
+
+	const MAX_ROWS_PER_QUERY = 500;
+
+	const LOG_WIKIA_LOGGER = 'wikia_logger';
+	const LOG_OUTPUT = 'output';
+
+	const ACTION_START = 'start';
+	const ACTION_LOG = 'log';
+	const ACTION_FAIL = 'fail';
+	const ACTION_FINISH = 'finish';
+
+	// An arbitrary amount of time to sleep between updates of MAX_ROWS_PER_QUERY rows to the DB
+	// If this is too high, we spend all our time sleeping, if too low, we hammer the DB.  Likely
+	// a value of 1 second (or lower if we use usleep()) would be sufficient.
+	const DB_COOL_DOWN_SECONDS = 1;
+
+	protected $startTime = 0;
+	protected $requestData = null;
+	protected $actionConfirmed = false;
+
+	protected $oldUsername = '';
+	protected $newUsername = '';
+
+	protected $userId = 0;
+	protected $fakeUserId = 0;
+	protected $requestorId = 0;
+	protected $requestorName = '';
+	protected $reason = null;
+	protected $notifyUser;
+
+	protected $errors = [ ];
+	protected $warnings = [ ];
+
+	protected $currentTaskId = null;
+
+	protected $repeatRename = false;
+	protected $phalanxBlockId = 0;
+
+	/**
+	 * Creates new rename user process
+	 *
+	 * @param string $oldUsername Old username
+	 * @param string $newUsername New username
+	 * @param bool $confirmed Has the user confirmed all the warnings he got?
+	 * @param string $reason
+	 * @param bool $notifyUser Whether to notify the renamed user
+	 */
+	public function __construct( $oldUsername, $newUsername, $confirmed = false, $reason = null, $notifyUser = true ) {
+		global $wgUser;
+
+		$this->startTime = time();
+
+		// Save original request data
+		$this->requestData = new \stdClass();
+		$this->requestData->oldUsername = $oldUsername;
+		$this->requestData->newUsername = $newUsername;
+
+		$this->actionConfirmed = $confirmed;
+		$this->reason = $reason;
+		$this->requestorId = $wgUser ? $wgUser->getId() : 0;
+		$this->requestorName = $wgUser ? $wgUser->getName() : '';
+		$this->notifyUser = $notifyUser;
+	}
+
+	/**
+	 * Create a new UserRenameToolProcessLocal object from select context data.
+	 *
+	 * @param array $data
+	 * @return ProcessBase
+	 */
+	public static function newFromData( $data ) {
+		$o = new static( $data['rename_old_name'], $data['rename_new_name'], '', true );
+
+		$mapping = [
+			'userId' => 'rename_user_id',
+			'oldUsername' => 'rename_old_name',
+			'newUsername' => 'rename_new_name',
+			'fakeUserId' => 'rename_fake_user_id',
+			'requestorId' => 'requestor_id',
+			'requestorName' => 'requestor_name',
+			'phalanxBlockId' => 'phalanx_block_id',
+			'reason' => 'reason',
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "reason" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)